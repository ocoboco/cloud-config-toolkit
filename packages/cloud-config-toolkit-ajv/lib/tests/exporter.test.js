const Ajv = require('ajv');

const Exporter = require('../exporter');
const keywords = require('./fixtures/keywords');
const schema = require('./fixtures/schema');
const validConfig = require('./fixtures/config/valid.json');
const invalidConfig = require('./fixtures/config/invalid.json');

describe('Exporter', function() {
  test('exports valid configuration', function() {
    const exporter = new Exporter({
      schema,
      keywords
    });
    expect(exporter.export(validConfig)).toMatchSnapshot();
  });

  test('throws an error when configuration is invalid', function() {
    const exporter = new Exporter({
      schema,
      keywords
    });
    expect(function() {
      exporter.export(invalidConfig);
    }).toThrow('invalid');
  });
});