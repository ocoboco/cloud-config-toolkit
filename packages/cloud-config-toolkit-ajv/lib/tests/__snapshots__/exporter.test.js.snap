// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Exporter exports valid configuration 1`] = `
Object {
  "settings": Object {
    "hostname": "google.com",
    "port": 80,
  },
}
`;
